# Pandas Assignment #2

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('kc_house_data.csv')

In [84]:
df.shape

(21613, 21)

We will start by taking the houses built from 2000 onwards. We will put it in a seperate dataFrame named df_2000.

In [75]:
df_2000=df.loc[df['yr_built']>=2000].copy()

We will check how many houses we have left:

In [83]:
df_2000.shape

(4759, 21)

So there is 4759 houses left.

We need to find the most expensive house for each year. We will put all the results in a new dataFrame named df_exp.
According to that, we will start by putting all the year values in a separate list:
Note that we should remove the duplicates by applying the set function.

In [62]:
yr_built=list(set(df_2000['yr_built'].values))

Now we will create an empty dataFrame and concat the row of the most expensive house for each year:

In [47]:
df_exp=pd.DataFrame()

In [48]:
for y in yr_built:
    df_exp=pd.concat([df_exp, df_2000.loc[df_2000['yr_built']==y].nlargest(1,'price')], axis=0, ignore_index=True)

In [113]:
list_columns=['id','price','bedrooms','bathrooms','sqft_living','sqft_lot','floors','condition','grade','yr_built','waterfront','view'] #list of column to display

In [110]:
df_exp[list_columns]

id      price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   7397300170  3710000.0         4       3.50         5550     28078     2.0   
1   9208900037  6890000.0         6       7.75         9890     31374     2.0   
2   3225069065  3080000.0         4       5.00         4550     18641     1.0   
3   6065300370  4210000.0         5       6.00         7440     21540     2.0   
4   7738500731  4500000.0         5       5.50         6640     40014     2.0   
5   3625049042  3640000.0         5       6.00         5490     19897     2.0   
6   3025059093  3100000.0         5       5.25         5090     23669     2.0   
7   4389201095  3650000.0         5       3.75         5020      8694     2.0   
8   6447300265  4000000.0         4       5.50         7080     16573     2.0   
9   8907500070  5350000.0         5       5.00         8000     23985     2.0   
10  1176001310  2950000.0         5       4.50         4340      5722     3.0   
11  5316100780  2580000.0         4       3.50         3280      3800     2.0   
12  1732800310  2000000.0         4       3.75         2870      4500     2.0   
13  8964800330  3000000.0         4       3.75         5090     14823     1.0   
14  2525049148  3420000.0         5       5.00         5450     20412     2.0   
15  2424059174  2000000.0         4       3.25         5640     35006     2.0   

    condition  grade  yr_built  waterfront  
0           4     12      2000           0  
1           3     13      2001           0  
2           3     10      2002           1  
3           3     12      2003           0  
4           3     12      2004           1  
5           3     12      2005           0  
6           3     12      2006           0  
7           3     12      2007           0  
8           3     12      2008           0  
9           3     12      2009           0  
10          3     10      2010           0  
11          3     11      2011           0  
12          3     10      2012           0  
13          3     11      2013           0  
14          3     11      2014           0  
15          3     11      2015           0

We need to find the houses for the most bedroom, bathroom and floors respectively. We will put them in seperate dataFrames named respectively df_maxbd , df_maxbth and df_maxflr

In [64]:
df_maxbd=pd.DataFrame()
df_maxbth=pd.DataFrame()
df_maxflr=pd.DataFrame()

In [65]:
for y in yr_built:
    df_maxbd=pd.concat([df_maxbd, df_2000.loc[df_2000['yr_built']==y].nlargest(1,'bedrooms')], axis=0, ignore_index=True)
    df_maxbth=pd.concat([df_maxbth, df_2000.loc[df_2000['yr_built']==y].nlargest(1,'bathrooms')], axis=0, ignore_index=True)
    df_maxflr=pd.concat([df_maxflr, df_2000.loc[df_2000['yr_built']==y].nlargest(1,'floors')], axis=0, ignore_index=True)

We will display the results below:

In [88]:
df_maxbd[list_columns] # The dataFrame of the houses with most bedrooms for each year starting 2000

id      price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   4139500200  1380000.0         6       4.50         5740     10312     2.0   
1    424069271   936000.0         7       3.75         5100     21802     2.0   
2   3425059222  1300000.0         6       3.50         6563     32670     2.0   
3   7202331220   635000.0         6       2.50         3880      5700     2.0   
4   1778360150  1240000.0         7       5.50         6630     13782     2.0   
5   7852011070  1140000.0         6       3.75         5960     20197     2.0   
6   3021059304   300000.0         6       3.00         2744      9926     2.0   
7   3185600055   495000.0         6       5.00         3440      4500     2.0   
8    627300145  1150000.0        10       5.25         4590     10920     1.0   
9   2149800278   343000.0         6       5.00         2732      7655     2.0   
10  1853081000   820000.0         5       2.75         2830      6137     2.0   
11  6891100260   830000.0         5       3.50         3700      5400     2.0   
12  5609000311   729999.0         6       4.50         3600      6110     2.0   
13  4006000183   450000.0         7       4.00         3150      7800     2.0   
14  2402100575  1130000.0         6       3.75         3010      4360     2.0   
15  9520900210   614285.0         5       2.75         2730      6401     2.0   

    condition  grade  yr_built  
0           3     11      2000  
1           3     10      2001  
2           3     10      2002  
3           3      7      2003  
4           3     10      2004  
5           3     10      2005  
6           3      7      2006  
7           3      8      2007  
8           3      9      2008  
9           3      7      2009  
10          3      9      2010  
11          3      9      2011  
12          3      9      2012  
13          3      8      2013  
14          3      9      2014  
15          3      8      2015

In [89]:
df_maxbth[list_columns] # The dataFrame of the houses with most bathrooms for each year starting 2000

id      price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   2626069030  1940000.0         4       5.75         7220    223462     2.0   
1   9208900037  6890000.0         6       7.75         9890     31374     2.0   
2   3225069065  3080000.0         4       5.00         4550     18641     1.0   
3   6065300370  4210000.0         5       6.00         7440     21540     2.0   
4   1630700380  1920000.0         5       5.75         7730    230868     2.0   
5   3625049042  3640000.0         5       6.00         5490     19897     2.0   
6    622069006  1500000.0         4       5.50         6550    217374     1.0   
7    424069279  1180000.0         6       6.50         6260     10955     2.0   
8   6447300265  4000000.0         4       5.50         7080     16573     2.0   
9   8907500070  5350000.0         5       5.00         8000     23985     2.0   
10  2524069097  2240000.0         5       6.50         7270    130017     2.0   
11  2619950060   465000.0         5       4.00         3210      7200     2.0   
12    42000130   600000.0         5       4.50         4440      9784     2.0   
13   524059063  1800000.0         5       5.00         4490     10279     2.0   
14  2525049148  3420000.0         5       5.00         5450     20412     2.0   
15  7853440050   771005.0         5       4.50         4000      6713     2.0   

    condition  grade  yr_built  
0           3     12      2000  
1           3     13      2001  
2           3     10      2002  
3           3     12      2003  
4           3     12      2004  
5           3     12      2005  
6           3     11      2006  
7           3     11      2007  
8           3     12      2008  
9           3     12      2009  
10          3     12      2010  
11          3      8      2011  
12          3     10      2012  
13          3     10      2013  
14          3     11      2014  
15          3      9      2015

In [92]:
df_maxflr[list_columns] # The dataFrame of the houses with most floors for each year starting 2000

id      price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   9528102772   438000.0         2       2.00         1270      1372     3.0   
1   7821200390   450000.0         3       2.00         1290      1213     3.0   
2   3300701615   655000.0         4       2.50         2630      4000     3.0   
3   9528104108   535000.0         3       2.50         1360      1016     3.0   
4   6798100662   312000.0         3       1.50         1255      1374     3.0   
5   1972202010   435000.0         3       3.00         1440      1350     3.5   
6   8673400177   525000.0         3       3.00         1730      1074     3.5   
7   1972200426   525000.0         2       2.75         1310      1268     3.5   
8   1702900664   479000.0         2       2.50         1730      1037     3.5   
9   4358700186   275000.0         3       2.25         1260      1488     3.0   
10  3362400431   518500.0         3       3.50         1590      1102     3.0   
11  2559950110  1230000.0         2       2.50         2470       609     3.0   
12  1112000031   715000.0         3       2.25         1990      4977     3.0   
13  5379804150   598800.0         6       4.00         4470     17877     3.0   
14  9528102996   549000.0         3       1.75         1540      1044     3.0   
15  9385200045   729500.0         3       2.50         1660      1091     3.0   

    condition  grade  yr_built  
0           3      8      2000  
1           3      8      2001  
2           3      8      2002  
3           3      7      2003  
4           3      7      2004  
5           3      8      2005  
6           3      8      2006  
7           3      8      2007  
8           3      8      2008  
9           3      7      2009  
10          3      8      2010  
11          3     11      2011  
12          3      9      2012  
13          3      9      2013  
14          3      8      2014  
15          3      9      2015

Next, we will find all the houses with an area larger that the mean area of all the houses. 

Accordingly we will store the mean value of the sqft_living and sqft_lot in 2 variables li_mean and lo_mean:

In [72]:
li_mean=df_2000['sqft_living'].mean()
lo_mean=df_2000['sqft_lot'].mean()

In [78]:
li_mean

2471.24479932759

In [125]:
lo_mean

12238.94788821181

First we will look for the houses with a sqft_living area larger that li_mean: we will store it into a dataframe called df_li:

In [124]:
df_li=df_2000.loc[df_2000['sqft_living']>li_mean].copy()

In [123]:
df_li

id             date      price  bedrooms  bathrooms  \
5      7237550310  20140512T000000  1230000.0         4       4.50   
29     1873100390  20150302T000000   719000.0         4       2.50   
40     5547700270  20140715T000000   625000.0         4       2.50   
42     7203220400  20140707T000000   861990.0         5       2.75   
45     8035350320  20140718T000000   488000.0         3       2.50   
...           ...              ...        ...       ...        ...   
21598  8956200760  20141013T000000   541800.0         4       2.50   
21599  7202300110  20140915T000000   810000.0         4       3.00   
21600   249000205  20141015T000000  1540000.0         5       3.75   
21605  3448900210  20141014T000000   610685.0         4       2.50   
21606  7936000429  20150326T000000  1010000.0         4       3.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
5             5420    101930     1.0           0     0  ...     11   
29            2570      7173     2.0           0     0  ...      8   
40            2570      5520     2.0           0     0  ...      9   
42            3595      5639     2.0           0     0  ...      9   
45            3160     13603     2.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21598         3118      7866     2.0           0     2  ...      9   
21599         3990      7838     2.0           0     0  ...      9   
21600         4470      8088     2.0           0     0  ...     11   
21605         2520      6023     2.0           0     0  ...      9   
21606         3510      7200     2.0           0     0  ...      9   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
5            3890           1530      2001             0    98053  47.6561   
29           2570              0      2005             0    98052  47.7073   
40           2570              0      2000             0    98074  47.6145   
42           3595              0      2014             0    98053  47.6848   
45           3160              0      2003             0    98019  47.7443   
...           ...            ...       ...           ...      ...      ...   
21598        3118              0      2014             0    98001  47.2931   
21599        3990              0      2003             0    98053  47.6857   
21600        4470              0      2008             0    98004  47.6321   
21605        2520              0      2014             0    98056  47.5137   
21606        2600            910      2009             0    98136  47.5537   

          long  sqft_living15  sqft_lot15  
5     -122.005           4760      101930  
29    -122.110           2630        6026  
40    -122.027           2470        5669  
42    -122.016           3625        5639  
45    -121.977           3050        9232  
...        ...            ...         ...  
21598 -122.264           2673        6500  
21599 -122.046           3370        6814  
21600 -122.200           2780        8964  
21605 -122.167           2520        6023  
21606 -122.398           2050        6200  

[2157 rows x 21 columns]

Now we will look for the houses with a sqft_lot area larger that li_lot: we will store it into a dataframe called df_lo:

In [122]:
df_lo=df_2000.loc[df_2000['sqft_lot']>lo_mean].copy()

In [121]:
df_lo

id             date      price  bedrooms  bathrooms  \
5      7237550310  20140512T000000  1230000.0         4        4.5   
45     8035350320  20140718T000000   488000.0         3        2.5   
79     1531000030  20150323T000000   720000.0         4        2.5   
114    8961960160  20141028T000000   480000.0         4        2.5   
147    3224079105  20140806T000000   430000.0         2        2.5   
...           ...              ...        ...       ...        ...   
21532   324069112  20140617T000000  1330000.0         4        4.0   
21540   715010530  20150113T000000  1880000.0         5        3.5   
21548  8835770330  20140819T000000  1060000.0         2        1.5   
21551  1561750040  20141224T000000  1380000.0         5        4.5   
21553  2522059251  20150409T000000   465000.0         3        2.5   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
5             5420    101930     1.0           0     0  ...     11   
45            3160     13603     2.0           0     0  ...      8   
79            3450     39683     2.0           0     0  ...     10   
114           3230     16171     2.0           0     3  ...      9   
147           2420     60984     2.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21532         4420     16526     2.0           0     0  ...     11   
21540         4410     13000     2.0           0     3  ...     10   
21548         2370    184231     2.0           0     0  ...     11   
21551         4350     13405     2.0           0     0  ...     11   
21553         2050     15035     2.0           0     0  ...      9   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
5            3890           1530      2001             0    98053  47.6561   
45           3160              0      2003             0    98019  47.7443   
79           3450              0      2002             0    98010  47.3420   
114          2520            710      2001             0    98001  47.3183   
147          2420              0      2007             0    98027  47.5262   
...           ...            ...       ...           ...      ...      ...   
21532        4420              0      2013             0    98075  47.5914   
21540        2920           1490      2014             0    98006  47.5382   
21548        2370              0      2005             0    98045  47.4543   
21551        4350              0      2014             0    98074  47.6018   
21553        2050              0      2006             0    98042  47.3619   

          long  sqft_living15  sqft_lot15  
5     -122.005           4760      101930  
45    -121.977           3050        9232  
79    -122.025           3350       39750  
114   -122.253           2640        8517  
147   -121.943           1940      193842  
...        ...            ...         ...  
21532 -122.027           3510       50447  
21540 -122.111           5790       12969  
21548 -121.778           3860      151081  
21551 -122.060           3990        7208  
21553 -122.122           1300       15836  

[460 rows x 21 columns]

In [120]:
df_lo['price'].mean()

1077252.8782608695

In [119]:
df_lo['price'].median()

836500.0

In [118]:
df_li['price'].mean()

841047.5187760779

In [117]:
df_li['price'].median()

725000.0

In [116]:
df_exp[list_columns]

id      price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   7397300170  3710000.0         4       3.50         5550     28078     2.0   
1   9208900037  6890000.0         6       7.75         9890     31374     2.0   
2   3225069065  3080000.0         4       5.00         4550     18641     1.0   
3   6065300370  4210000.0         5       6.00         7440     21540     2.0   
4   7738500731  4500000.0         5       5.50         6640     40014     2.0   
5   3625049042  3640000.0         5       6.00         5490     19897     2.0   
6   3025059093  3100000.0         5       5.25         5090     23669     2.0   
7   4389201095  3650000.0         5       3.75         5020      8694     2.0   
8   6447300265  4000000.0         4       5.50         7080     16573     2.0   
9   8907500070  5350000.0         5       5.00         8000     23985     2.0   
10  1176001310  2950000.0         5       4.50         4340      5722     3.0   
11  5316100780  2580000.0         4       3.50         3280      3800     2.0   
12  1732800310  2000000.0         4       3.75         2870      4500     2.0   
13  8964800330  3000000.0         4       3.75         5090     14823     1.0   
14  2525049148  3420000.0         5       5.00         5450     20412     2.0   
15  2424059174  2000000.0         4       3.25         5640     35006     2.0   

    condition  grade  yr_built  waterfront  view  
0           4     12      2000           0     2  
1           3     13      2001           0     4  
2           3     10      2002           1     4  
3           3     12      2003           0     0  
4           3     12      2004           1     4  
5           3     12      2005           0     0  
6           3     12      2006           0     0  
7           3     12      2007           0     1  
8           3     12      2008           0     0  
9           3     12      2009           0     4  
10          3     10      2010           0     4  
11          3     11      2011           0     0  
12          3     10      2012           0     3  
13          3     11      2013           0     0  
14          3     11      2014           0     0  
15          3     11      2015           0     2

In [115]:
df_maxbd[list_columns]

id      price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   4139500200  1380000.0         6       4.50         5740     10312     2.0   
1    424069271   936000.0         7       3.75         5100     21802     2.0   
2   3425059222  1300000.0         6       3.50         6563     32670     2.0   
3   7202331220   635000.0         6       2.50         3880      5700     2.0   
4   1778360150  1240000.0         7       5.50         6630     13782     2.0   
5   7852011070  1140000.0         6       3.75         5960     20197     2.0   
6   3021059304   300000.0         6       3.00         2744      9926     2.0   
7   3185600055   495000.0         6       5.00         3440      4500     2.0   
8    627300145  1150000.0        10       5.25         4590     10920     1.0   
9   2149800278   343000.0         6       5.00         2732      7655     2.0   
10  1853081000   820000.0         5       2.75         2830      6137     2.0   
11  6891100260   830000.0         5       3.50         3700      5400     2.0   
12  5609000311   729999.0         6       4.50         3600      6110     2.0   
13  4006000183   450000.0         7       4.00         3150      7800     2.0   
14  2402100575  1130000.0         6       3.75         3010      4360     2.0   
15  9520900210   614285.0         5       2.75         2730      6401     2.0   

    condition  grade  yr_built  waterfront  view  
0           3     11      2000           0     2  
1           3     10      2001           0     0  
2           3     10      2002           0     0  
3           3      7      2003           0     0  
4           3     10      2004           0     0  
5           3     10      2005           0     4  
6           3      7      2006           0     0  
7           3      8      2007           0     0  
8           3      9      2008           0     2  
9           3      7      2009           0     0  
10          3      9      2010           0     0  
11          3      9      2011           0     0  
12          3      9      2012           0     0  
13          3      8      2013           0     0  
14          3      9      2014           0     0  
15          3      8      2015           0     0

Looking at the results we can lift the following conclusions:
- The number of bedrooms, bathrooms and floors does not have a direct relation to the prices of the house since the gap between the most expensive houses and the others is high
- The living area size has a higher influence on the prices thant the lot area size
- The grade has influence on the prices: whenever there is 2 houses with close area size the grade does make a difference on the price
